pkg/gitops/flux/git.go

```diff
@@ -0,0 +1,100 @@
+package flux
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/git"
+	gitFactory "github.com/aws/eks-anywhere/pkg/git/factory"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+)
+
+type gitClient struct {
+	git         git.Client
+	gitProvider git.ProviderClient
+	*retrier.Retrier
+}
+
+func newGitClient(gitTools *gitFactory.GitTools) *gitClient {
+	if gitTools == nil {
+		return nil
+	}
+
+	return &gitClient{
+		git:         gitTools.Client,
+		gitProvider: gitTools.Provider,
+		Retrier:     retrier.NewWithMaxRetries(maxRetries, backOffPeriod),
+	}
+}
+
+func (c *gitClient) GetRepo(ctx context.Context) (repo *git.Repository, err error) {
```

Comment:
the git factor will return a `nil` provider for the generic git case; https://github.com/aws/eks-anywhere/blob/main/pkg/git/factory/gitfactory.go#L57 

it could be worth adding a nil check for gitTools.Provider in these methods to make that explicit